src/main/scala/mimir/models/SparkClassifierModel.scala

```diff
@@ -0,0 +1,221 @@
+package mimir.models
+
+import java.io.File
+import java.sql.SQLException
+import java.util
+import com.typesafe.scalalogging.slf4j.Logger
+
+import mimir.algebra._
+import mimir.ctables._
+import mimir.util.{RandUtils,TextUtils,TimeUtils}
+import mimir.{Analysis, Database}
+
+import mimir.models._
+
+import mimir.ml.spark.MultiClassClassification
+
+import scala.util._
+import org.apache.spark.sql.types.StringType
+
+object SparkClassifierModel
+{
+  val logger = Logger(org.slf4j.LoggerFactory.getLogger(getClass.getName))
+  val TRAINING_LIMIT = 10000
+  val TOKEN_LIMIT = 100
+
+  def train(db: Database, name: String, cols: Seq[String], query:Operator): Map[String,(Model,Int,Seq[Expression])] = 
+  {
+    cols.map( (col) => {
+      val modelName = s"$name:$col"
+      val model = 
+        db.models.getOption(modelName) match {
+          case Some(model) => model
+          case None => {
+            val model = new SimpleSparkClassifierModel(modelName, col, query)
+            model.train(db)
+            db.models.persist(model)
+            model
+          }
+        }
+
+      col -> (
+        model,                         // The model for the column
+        0,                             // The model index of the column's replacement variable
+        query.columnNames.map(Var(_))  // 'Hints' for the model -- All of the remaining column values
+      )
+    }).toMap
+  }
+
+  
+
+  
+}
+
+@SerialVersionUID(1000L)
+class SimpleSparkClassifierModel(name: String, colName: String, query: Operator)
+  extends Model(name) 
+  with NeedsReconnectToDatabase 
+{
+  val feedback = scala.collection.mutable.Map[String,PrimitiveValue]()
+  val classificationCache = scala.collection.mutable.Map[String,PrimitiveValue]()
+  val colIdx:Int = query.columnNames.indexOf(colName)
+  val classifyUpFrontAndCache = true
+  var learner: Option[MultiClassClassification.ClassifierModel] = None
+  
+  @transient var db: Database = null
+
+  
+  /**
+   * When the model is created, learn associations from the existing data.
+   */
+  def train(db:Database)
+  {
+    this.db = db
+    TimeUtils.monitor(s"Train $name.$colName", WekaModel.logger.info(_)){
+      val trainingQuery = Limit(0, Some(SparkClassifierModel.TRAINING_LIMIT), Project(Seq(ProjectArg(colName, Var(colName))), query))
+      learner = Some(MultiClassClassification.NaiveBayesMulticlassModel(MultiClassClassification.prepareValueStr _, MultiClassClassification.getSparkTypeStr _)(trainingQuery, db, colName))
```

Comment:
Does spark make it any easier to build models for data that may not be classifiable?  (e.g., continuous or prose data?)